internal/fs/inode/dir.go

```diff
@@ -1034,6 +1037,93 @@ func (d *dirInode) DeleteChildDir(
 	return nil
 }
 
+// LOCKS_REQUIRED(d)
+func (d *dirInode) DeleteObjects(ctx context.Context, objectNames []string) error {
+	for _, objectName := range objectNames {
+		if strings.HasSuffix(objectName, "/") {
+			// Initiate deletion for a prefix (directory). This logic handles pagination internally.
+			if err := d.deletePrefixRecursively(ctx, objectName); err != nil {
+				return fmt.Errorf("recursively deleting prefix %q: %w", objectName, err)
+			}
+		} else {
+			// Handle single file-like object deletion.
+			if err := d.deleteSingleObject(ctx, objectName); err != nil {
+				return fmt.Errorf("deleting unsupported object %q: %w", objectName, err)
+			}
+		}
+	}
+	return nil
+}
+
+// Helper to delete a single object, handling 'Not Found' errors gracefully.
```

Comment:
what is the need to handle NotFound errors gracefully? can you add a comment here.